### PR TITLE
Support `SHOW VITESS_MIGRATIONS` from inside a transaction

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

See story in https://github.com/vitessio/vitess/issues/16242

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/16242

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by systay** [comment]
Could we add an end-to-end test that checks this issue?

**Comment by shlomi-noach** [comment]
> Could we add an end-to-end test that checks this issue?

Added!

**Review by systay — APPROVED** [review]

File: go/vt/vttablet/tabletserver/query_executor.go [hunk 1]
```diff
@@ -958,7 +960,7 @@ func (qre *QueryExecutor) execRevertMigration() (*sqltypes.Result, error) {
 	return qre.tsv.onlineDDLExecutor.SubmitMigration(qre.ctx, qre.plan.FullStmt)
 }
 
-func (qre *QueryExecutor) execShowMigrations() (*sqltypes.Result, error) {
+func (qre *QueryExecutor) execShowMigrations(conn *StatefulConnection) (*sqltypes.Result, error) {
```
Comment (timvaillancourt) [inline go/vt/vttablet/tabletserver/query_executor.go:963]: @shlomi-noach is `conn *StatefulConnection` meant to be used here? I don't follow this part

Comment (shlomi-noach) [reply go/vt/vttablet/tabletserver/query_executor.go:963]: Yes. I'm not all too familiar with this either, but I got guidance from @harshit-gangal. The `StatefulConnection` is the standard way for you to implement a transactional behavior working with QueryService. So we should pass that variable. Normally I'd also _use_ the variable -- except:
- I don't really require `show vitess_migrations` to exhibit transactional behavior; it's a `show` command and I don;t need it to e.g. be repeatable-read or anything. I only need it to work from within a transaction.
- Therefore no need to bloat the transaction.
- Also, the existing logic uses the executor's own connection pooling, which is fine, and there's no need/reason to now take apart the existing and working mechanism within the executor to accommodate the transactional connection.
-

Comment (timvaillancourt) [reply go/vt/vttablet/tabletserver/query_executor.go:963]: Ok sounds good 👍

**Review by timvaillancourt — APPROVED** [review]

**Pull request merged**